### PR TITLE
Fix the StringToNumber Script Canvas node

### PR DESCRIPTION
**Pull request by junbo75**
Signed-off-by: Junbo Liang <68558268+junbo75@users.noreply.github.com>

Found an issue with the StringToNumber Script Canvas node where it doesn't accept any input. Got the following error when added this node to a graph:
`[Error] (ScriptCanvas) - NodeFunctionGenericMultiReturn failed to add a required data slot`

Verified that the node worked as expected with this PR.
Before:
Before
After:
image

### CODE REVIEW

File: Gems/ScriptCanvas/Code/Include/ScriptCanvas/Libraries/Math/MathGenerics.h [hunk 1]
```diff
@@ -28,7 +28,7 @@ namespace ScriptCanvas
             return AZStd::stof(stringValue);
         }
 
-        SCRIPT_CANVAS_GENERIC_FUNCTION_NODE(StringToNumber, k_categoryName, "{FD2D9758-5EA2-45A3-B293-A748D951C4A3}", "Converts the given string to it's numeric representation if possible.", "", "");
+        SCRIPT_CANVAS_GENERIC_FUNCTION_NODE(StringToNumber, k_categoryName, "{FD2D9758-5EA2-45A3-B293-A748D951C4A3}", "Converts the given string to it's numeric representation if possible.", "StringValue");
```
Comment (onecent1101) [inline Gems/ScriptCanvas/Code/Include/ScriptCanvas/Libraries/Math/MathGenerics.h:31]: Can you provide before&after screenshot to show what has been changed?

And why do you need this change here? I think previously two quotes are for output and input?

**Review by onecent1101 — COMMENTED** [review]
can you also provide what error you got before this change to help understand the issue?
based on this change, seems like just for node naming?

File: Gems/ScriptCanvas/Assets/TranslationAssets/Nodes/Math_StringToNumber.names [hunk 2]
```diff
@@ -27,6 +27,12 @@
                     "details": {
                         "name": "Result"
                     }
+                },
+                {
+                    "base": "DataInput_StringValue_0",
+                    "details": {
+                        "name": "StringValue"
```
Comment (lsemp3d) [inline Gems/ScriptCanvas/Assets/TranslationAssets/Nodes/Math_StringToNumber.names:34]: Please add a space (i.e. "String Value") to be consistent with other parameter names

**Comment by lsemp3d** [comment]
> can you also provide what error you got before this change to help understand the issue? based on this change, seems like just for node naming?

@junbo75 yes, please add the callstack of the crash, I'd like to see the root cause to prevent future crashes due to this issue, thank you!

File: Gems/ScriptCanvas/Code/Include/ScriptCanvas/Libraries/Math/MathGenerics.h (hunk 1 above)
Comment (junbo75) [reply Gems/ScriptCanvas/Code/Include/ScriptCanvas/Libraries/Math/MathGenerics.h:31]: Sure, I can provide the screenshots for this node with/without this change. Based on the definition for other nodes using the SCRIPT_CANVAS_GENERIC_FUNCTION_NODE, the quotes after node description should be the node inputs.

#define SCRIPT_CANVAS_GENERIC_FUNCTION_NODE(NODE_NAME, CATEGORY, UUID, DESCRIPTION, ...)\
    SCRIPT_CANVAS_GENERIC_FUNCTION_MULTI_RESULTS_NODE_WITH_DEFAULTS(NODE_NAME, ScriptCanvas::NoDefaultArguments, CATEGORY, UUID, false, DESCRIPTION, __VA_ARGS__)

**Comment by junbo75** [comment]
> > can you also provide what error you got before this change to help understand the issue? based on this change, seems like just for node naming?
> 
> @junbo75 yes, please add the callstack of the crash, I'd like to see the root cause to prevent future crashes due to this issue, thank you!

I didn't see any crash caused by the issue. Without this PR, when I drag a StringToNumber node to a graph, the node doesn't accept any input and is not actually usable. I'll upload screenshots for this node in the PR description.

> > can you also provide what error you got before this change to help understand the issue? based on this change, seems like just for node naming?
> 
> @junbo75 yes, please add the callstack of the crash, I'd like to see the root cause to prevent future crashes due to this issue, thank you!

I didn't see any crash caused by this issue. When I dragged the StringToNumber node to a graph, the node didn't accept a string input and wasn't actually usable. I just uploaded the screenshots with/without this change.

File: Gems/ScriptCanvas/Code/Include/ScriptCanvas/Libraries/Math/MathGenerics.h (hunk 1 above)
Comment (onecent1101) [reply Gems/ScriptCanvas/Code/Include/ScriptCanvas/Libraries/Math/MathGenerics.h]: actually the last part of arguments are for inputs and outputs name together, so do you really need this change, seems like you just need to add input name translation

Comment (onecent1101) [reply Gems/ScriptCanvas/Code/Include/ScriptCanvas/Libraries/Math/MathGenerics.h]: confirmed locally the issue is caused by trying to add slot with empty name
https://github.com/o3de/o3de/blob/development/Gems/ScriptCanvas/Code/Include/ScriptCanvas/Core/Node.cpp#L2287

Fix is to give slot name, translation change is optional but not harm to have

**Review by onecent1101 — APPROVED** [review]

**Review by AMZN-alexpete — APPROVED** [review]

**Pull request merged**